jobs/build.Jenkinsfile

```
@@ -347,6 +347,17 @@ lock(resource: "build-${params.STREAM}") {
         // Build the remaining artifacts
         stage("Build Artifacts") {
             pipeutils.build_artifacts(pipecfg, params.STREAM, basearch)
+
+            // Stop the build if the kernel + kernel-rt versions do not match
+            if (pipecfg.streams[params.STREAM]?.check_kernel_rt_mismatch_rhcos == true) {
+                echo("Verifying kernel + kernel-rt versions match")
+                build_meta = [readJSON(file: "builds/latest/${basearch}/commitmeta.json"), readJSON(file: "builds/latest/${basearch}/meta.json")]
+                kernel_version = build_meta[0]['ostree.linux'].split('.el')[0]
+                kernel_rt_version = [build_meta[1]['extensions']['manifest']['kernel-rt-core'].split('.rt')[0]
+                assert kernel_version == kernel_rt_version : "kernel-rt version: ${kernel_rt_version} does not match kernel version: ${kernel_version}"
+            } else {
+                echo("Skipping kernel + kernel-rt mismatch check")
+            }
```

```suggestion
            }
```

I would probably just skip saying anything if that was the case.

---

Works for me. That way we dont see this on every FCOS build.

---

Removed with latest push